community/blog/2020.10-Introducing-message-handling-with-Azure-Service-Bus-message-influence/README.md

```diff
@@ -0,0 +1,107 @@
+- **Dead lettering** a message which marks the Service Bus message 'dead lettered' on Azure so it's moved to the dead letter queue.
+- **Abandoning** a message which marks the Service Bus message 'abandonned' on Azure.
+
+## Specific Azure Service Bus message handlers
+
+Because of the way the message handlers are registered and used inside the application, there's no 'callback' function to access the Azure Service Bus.
+That's why we've created two 'message handler templates' to help with this. These templates implements the message handler signature but also provides you with a whole range of specific Azure Service Bus functions that are related to completing the message processing.
+
+- `AzureServiceBusMessageHandler<>`: is an `IMessageHandler` implementation which provides the specific Azure Service Bus message operations
+- `AzureServiceBusFallbackMessageHandler`: is an `IServiceBusFallbackMessageHandler` implementation which provides the specific Azure Service Bus message operations
+
+### Regular message handling
+
+Here is an example of how a normal `IMessageHandler` implementation is created for Azure Service Bus.
+The `IAzureServiceBusMessageHanlder<>` is used here as shortcut, which in fact implemnets the `IMessageHandler<>`.
+
+```csharp
+public class OrderMessageHandler<Order> : IAzureServiceBusMessageHandler<Order>
+{
+    public async Task ProcessMessageAsync(Order order, AzureServiceBusMessageContext context, ...)
+    {
+        // Processing order.
+    }
+}
+```
+
+When you decide you want to abandon the message when the order is not recognized or the message context is invalid, you'll have to implement `AzureServiceBusMessageHandler<>` instead.
+
+```csharp
+public class OrderMessageHandler<Order> : AzureServiceBusMessageHandler<Order>
+{
+    public override async Task ProcessMessageAsync(Order order, AzureServiceBusMessageContext context, ...)
+    {
+        if (order.Customer is null)
+        {
+            // Calling base operation where the Azure Service Bus operations are located.
+            await base.AbandonMessageAsync();
+        }
+        else
+        {
+            // Processing order.
+        }
+    }
+}
+```
+
+Note that in the regular message handling you don't have to specify which message to abandon. This is all done behind the screens so the message handler only has to focus on the message processing and call the specific Azure Service Bus message operations when necessary.
+
+### Fallback message handling
+
+The fallback message handling is possible more correct when we're talking about dead lettering a message. Fallback message handlers will process the Azure Service Bus message when none other regular message handler was able to correctly process it.
+At the end of the fallback message processing, the message could be dead lettered on Azure for example.
+
+Here's an example of how a normal `IServiceBusFallbackMessageHandler` is implemented:
```

Comment:
Maybe show a registration example in this section how it has 2 message handlers and then this one as a fallback?

Reply:
Ah, yes, that would be good.

Reply:
Added some.